### ISSUE TITLE
[TT-2388] - Adding APIs with Swagger doesn't correctly combine basePath and API paths

### ISSUE THREAD

**Issue by l3ender** [TykTechnologies/tyk#2443]
**Branch/Environment/Version**
- Branch/Version: v2.8.3
- Environment: on-prem

**Describe the bug**
When adding APIs from a Swagger definition, the created API endpoints are not separated by a slash, leading to an incorrect URL.

**Reproduction steps**

1. Add API from Swagger JSON. Example below.

**Actual behavior**
The API endpoint URL is generated in the format `/base_pathendpoint_path` (no separating `/`).

**Expected behavior**
The API endpoint URL should be generated correctly in the format `/base_path/endpoint_path`.

**Screenshots/Video**
<img width="400" alt="Screen Shot 2019-08-01 at 4 28 31 PM" src="https://user-images.githubusercontent.com/2243641/62329158-b5c35d00-b47a-11e9-81fe-26cdbe200992.png">
<img width="400" alt="Screen Shot 2019-08-01 at 4 28 49 PM" src="https://user-images.githubusercontent.com/2243641/62329170-ba881100-b47a-11e9-9ca0-7c32e87d07ba.png">

**Logs (debug mode or log file):**
<details>
    <summary>Gateway log detail</summary>

```
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Initialising Tyk REST API Endpoints"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Loading API configurations."
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Tracking hostname" api_name="New Application REST API" domain="(no host)"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Loading API" api_id=f3ecc4d5097c45a37a10f737e990008b api_name="New Application REST API" org_id=5caf43527871d30001813bc6
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Checking security policy: Open" api_id=f3ecc4d5097c45a37a10f737e990008b api_name="New Application REST API" org_id=5caf43527871d30001813bc6
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="API Loaded" api_id=f3ecc4d5097c45a37a10f737e990008b api_name="New Application REST API" org_id=5caf43527871d30001813bc6 server_name=-- user_id=-- user_ip=--
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Processed and listening on: /new-application-api{rest:.*}"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Initialised API Definitions"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="API reload complete"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="reload: complete"
tyk-gateway_1    | time="Aug  1 21:41:26" level=info msg="Initiating coprocess reload"
```

</details>

**Configuration (tyk config file):**
Default installation.

**Additional context**
If the `basePath` value in Swagger definition has a trailing slash, everything works as expected. However, the [Swagger spec](https://swagger.io/docs/specification/2-0/api-host-and-base-path/) shows that basePath should not have a trailing slash.

**Comment by letzya** [comment]
@ilijabojanovic there's a fix for this right?

**Comment by ilijabojanovic** [comment]
@l3ender Could you please attach swagger spec and generated api definition?

**Comment by l3ender** [comment]
My apologies. Here it is: 

<details><summary>new-application-api.json</summary>

```json
{
	"swagger": "2.0",
	"info": { "version": "1.0.0", "title": "New Application REST API" },
	"host": "example.com",
	"basePath": "/new-application-api",
	"tags": [{ "name": "Application", "description": "Application Controller" }],
	"paths": {
		"/application": {
			"post": {
				"tags": ["Application"],
				"summary": "Create an application",
				"operationId": "createApplicationUsingPOST",
				"consumes": ["application/json"],
				"produces": ["*/*"],
				"parameters": [
					{
						"in": "body",
						"name": "createApplicationRequest",
						"description": "createApplicationRequest",
						"required": true,
						"schema": { "$ref": "#/definitions/CreateApplicationRequest" }
					}
				],
				"responses": {
					"200": {
						"description": "OK",
						"schema": { "$ref": "#/definitions/CreateApplicationResponse" }
					},
					"201": { "description": "Created" },
					"401": { "description": "Unauthorized" },
					"403": { "description": "Forbidden" },
					"404": { "description": "Not Found" }
				},
				"deprecated": false
			}
		},
		"/application/search": {
			"post": {
				"tags": ["Search Applications"],
				"summary": "searchApplication",
				"operationId": "searchApplicationUsingPOST",
				"consumes": ["application/json"],
				"produces": ["*/*"],
				"parameters": [
					{
						"in": "body",
						"name": "request",
						"description": "request",
						"required": true,
						"schema": { "$ref": "#/definitions/ApplicationSearchRequest" }
					}
				],
				"responses": {
					"200": {
						"description": "OK",
						"schema": { "$ref": "#/definitions/ApplicationSearchResponse" }
					},
					"201": { "description": "Created" },
					"401": { "description": "Unauthorized" },
					"403": { "description": "Forbidden" },
					"404": { "description": "Not Found" }
				},
				"deprecated": false
			}
		}
	},
	"definitions": {
		"Applicant": {
			"type": "object",
			"properties": {
				"applicantId": { "type": "string" },
				"applicantType": { "type": "string" },
				"createdDateTime": { "type": "string", "format": "date-time" },
				"ssnLastFour": { "type": "string" },
				"updatedDateTime": { "type": "string", "format": "date-time" }
			},
			"title": "Applicant"
		},
		"BasicApplicant": {
			"type": "object",
			"properties": {
				"applicantId": { "type": "string" },
				"applicantType": { "type": "string" },
				"dateOfBirth": { "type": "string", "format": "date-time" },
				"housingStatus": { "type": "string" },
				"relationToPrimary": { "type": "string" }
			},
			"title": "BasicApplicant"
		},
		"CreateApplicationRequest": {
			"type": "object",
			"properties": {
				"applicants": {
					"type": "array",
					"items": { "$ref": "#/definitions/BasicApplicant" }
				},
				"applicationType": { "type": "string" }
			},
			"title": "CreateApplicationRequest"
		},
		"CreateApplicationResponse": {
			"type": "object",
			"properties": {
				"applicants": {
					"type": "array",
					"items": { "$ref": "#/definitions/Applicant" }
				},
				"applicationId": { "type": "string" }
			},
			"title": "CreateApplicationResponse"
		}
	}
}

```

</details>

**Comment by christtyk** [comment]
@ilijabojanovic do you have any update on this one?

**Comment by christtyk** [comment]
Needs to stay open - client is having to implement an inelegant manual workaround

**Comment by l3ender** [comment]
Keep open.

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 4.0.6 (LTS) and Tyk 4.2.1.

I'm not sure why this ticket wasn't updated, apologies for the delay.
